ftl-build/Dockerfile

```diff
@@ -45,9 +45,12 @@ RUN apk add --no-cache \
 ENV STATIC true
 ENV TEST true
 
+# As of Alpine 3.21 (Dec 2024), we need to patch the termcap library to include
+# the standard headers as well as unistd.h for the write function
 RUN curl -sSL https://ftl.pi-hole.net/libraries/termcap-${termcapversion}.tar.gz | tar -xz \
     && cd termcap-${termcapversion} \
     && ./configure --enable-static --disable-shared --disable-doc --without-examples \
+    && sed -i '1i #define STDC_HEADERS 1\n#include <unistd.h>' termcap.c tparam.c \
```

Comment:
Patching is necessary due to a [breaking change in GCC 14](https://gcc.gnu.org/gcc-14/porting_to.html#warnings-as-errors):

> **Implicit function declarations (`-Werror=implicit-function-declaration`)**
> It is no longer possible to call a function that has not been declared. In general, the solution is to include a header file with an appropriate function prototype. Note that GCC will perform further type checks based on the function prototype, which can reveal further type errors that require additional changes.
> 
> For well-known functions declared in standard headers, GCC provides fix-it hints with the appropriate `#include` directives:
> ```
>error: implicit declaration of function ‘strlen’ [-Wimplicit-function-declaration]
>    5 |   return strlen (s);
>      |          ^~~~~~
>note: include ‘<string.h>’ or provide a declaration of ‘strlen’
>  +++ |+#include <string.h>
>    1 |
> ```
> On GNU systems, headers described in standards (such as the C standard, or POSIX) may require the definition of certain macros at the start of the compilation before all required function declarations are made available. See [Feature Test Macros](https://sourceware.org/glibc/manual/latest/html_node/Feature-Test-Macros.html) in the GNU C Library manual for details.